src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -483,21 +491,46 @@ public void generateComponentLocationAnalysisIfEnabled(BdioResult bdio) throws O
      * @param rapidFullResults
      * @throws OperationException
      */
-    public void generateComponentLocationAnalysisIfEnabled(List<DeveloperScansScanView> rapidFullResults) throws OperationException {
+    public void generateComponentLocationAnalysisIfEnabled(List<DeveloperScansScanView> rapidFullResults, BdioResult bdio) throws OperationException {
         if (detectConfigurationFactory.isComponentLocationAnalysisEnabled()) {
-            if (!rapidFullResults.isEmpty()) {
-                auditLog.namedPublic(
-                        "Generate Component Location Analysis File for Reported Components",
-                        () -> (new GenerateComponentLocationAnalysisOperation()).locateComponentsForNonPersistentOnlineDetectorScan(rapidFullResults, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
-                );
-            } else {
+            if (applicableDetectorsIncludeAtLeastOneSupportedDetector(bdio.getApplicableDetectorTypes())) {
+                if (!rapidFullResults.isEmpty()) {
+                    auditLog.namedPublic(
+                            "Generate Component Location Analysis File for Reported Components",
+                            () -> (new GenerateComponentLocationAnalysisOperation()).locateComponentsForNonPersistentOnlineDetectorScan(rapidFullResults, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
+                    );
+                } else {
+                    logger.info(ReportConstants.RUN_SEPARATOR);
+                    logger.info("Component Location Analysis requires non-empty Rapid Scan results. Skipping location analysis.");
+                    logger.info(ReportConstants.RUN_SEPARATOR);
+                }
+        } else {
                 logger.info(ReportConstants.RUN_SEPARATOR);
-                logger.info("Component Location Analysis requires non-empty Rapid Scan results. Skipping location analysis.");
+                logger.info("Component Location Analysis only supports NPM, Maven, Gradle and NuGet detectors.");
                 logger.info(ReportConstants.RUN_SEPARATOR);
             }
         }
     }
 
+    private boolean applicableDetectorsIncludeAtLeastOneSupportedDetector(Set<DetectorType> applicableDetectors) {
+        applicableDetectors.retainAll(SUPPORTED_DETECTORS);
+        return !applicableDetectors.isEmpty();
```

Comment:
I see that the `applicableDetectors` set is coming from the `BdioResult` class. Is the `applicableDetectors` set being modified after the bdio result has been written to file? I'm a little concerned about updating the BdioResult class's set (since we are passing the set by reference and updating it) in this CLL post-processing workflow.
As I understand, `applicableDetectors` here is in context of CLL only. Would it not be safer then to create a new copy of `bdio.getApplicableDetectors` and update it everywhere in the CLL workflow?
I think this allows us to keep BdioResult intact and it's data can be used by any other post-processing operation in the future. Let me know if I misunderstood the code somewhere.

Reply:
Please see the changes in this commit, I believe that will answer the above concerns as we are no longer modifying applicableDetectors set but instead create an equivalent Set<String> using the names of the DetectorType's instead.

Reply:
Resolving convo as per offline discussion where we clarified the reason for the changes in BdioResult class.